packages/next/client/route-announcer.tsx

```diff
@@ -36,14 +36,9 @@ export const RouteAnnouncer = () => {
       if (previouslyLoadedPath.current === asPath) return
       previouslyLoadedPath.current = asPath
 
-      if (document.title) {
-        setRouteAnnouncement(document.title)
-      } else {
-        const pageHeader = document.querySelector('h1')
-        const content = pageHeader?.innerText ?? pageHeader?.textContent
+      const pageHeader = document.querySelector('h1')
```

Comment:
Running DOM API takes time. Why do you query it first even if `document.title` is already available?

You should always write the code **without** tricks, and just let modification does its jobs.

Reply:
Understood :)